src/app/modules/todo/services/todo.service.ts

```diff
@@ -2,22 +2,29 @@ import { Injectable } from '@angular/core';
 import { TodoStore } from '../models/todo.store';
 import { Todo, SearchObject } from '../models/todo.model';
 import { AngularFirestore } from '@angular/fire/firestore';
-import { map } from 'rxjs/operators';
-import { Observable } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+import { AuthService } from 'src/app/common/services/auth.service';
+import { StatusRequest } from 'src/app/common/models/auth.model';
 
 @Injectable({ providedIn: 'root' })
 export class TodoService {
   constructor(
     private todoStore: TodoStore,
-    private firestore: AngularFirestore
+    private firestore: AngularFirestore,
+    private authService: AuthService
   ) {}
 
   public getTodo(): Observable<Todo[]> {
     return this.firestore
       .collection('Todo')
       .snapshotChanges()
       .pipe(
-        map((result) => {
+        switchMap((result) => {
+          return this.authService.checkAuthAPI(result);
+        }),
+        map((result: any[]) => {
+          console.log(result);
           const listTodo: Todo[] = this.formatListTodo(result);
```

Comment:
you don't need identity type of listTodo variable because typescript compiler will understand that the type of variable will be return type of this function `formatListTodo`